src/main/java/com/exactpro/th2/common/schema/filter/strategy/impl/AbstractFilterStrategy.java

```diff
@@ -57,6 +57,7 @@ private MultiValuedMap<String, FieldFilterConfiguration> extractConfigMapFromRou
         MultiValuedMap<String, FieldFilterConfiguration> msgFieldFilters = MultiMapUtils.newListValuedHashMap();
         msgFieldFilters.putAll(routerFilter.getMessage());
         msgFieldFilters.putAll(routerFilter.getMetadata());
```

Comment:
Different maps can contain the same kays. We should check each filter vs a suitable part of a message